src/context/alert.tsx

```diff
@@ -42,96 +49,113 @@ export enum ERROR_MESSAGE {
   AUTH_FAILURE_TAKEN_TOKEN = "Get a new QR code that is not tagged to any contact number from your in-charge.",
   OTP_EXPIRED = "Get a new OTP and try again.",
   LOGIN_ERROR = "We are currently facing login issues. Get a new QR code from your in-charge.",
-  PAST_TRANSACTIONS_ERROR = "We are currently facing server issues. Try again later or contact your in-charge if the problem persists."
+  PAST_TRANSACTIONS_ERROR = "We are currently facing server issues. Try again later or contact your in-charge if the problem persists.",
+  VALIDATE_INPUT_REGEX_ERROR = "Please check that the ID is in the correct format"
 }
 
-const defaultAlertProps: AlertModalProps = {
-  alertType: "ERROR",
-  title: "",
-  buttonTexts: {
-    primaryActionText: "OK"
-  },
-  visible: false,
-  onOk: () => {},
-  onCancel: () => {},
-  onExit: () => {}
+const errorNameToTranslationKeyMappings: Record<string, string> = {
+  CampaignConfigError: "systemErrorConnectivityIssues",
+  SessionError: "expiredQR",
+  PastTransactionError: "systemErrorServerIssues",
+  QuotaError: "systemErrorConnectivityIssues",
+  OTPWrongError: "invalidInputOTP",
+  OTPWrongErrorLastTry: "invalidInputOTPOneMoreInvalid",
+  OTPExpiredError: "expiredOTP",
+  LoginLockedError: "disabledAccess",
+  LoginError: "systemErrorLoginIssue",
+  AuthTakenError: "alreadyUsedQRCode",
+  AuthExpiredError: "expiredQR",
+  AuthNotFoundError: "expiredQR",
+  AuthInvalidError: "wrongFormatQRScanAgain",
+  ScannerError: "errorScanning",
+  LimitReachedError: "scanLimitReached",
+  NotEligibleError: "notEligible"
 };
 
-export const defaultWarningProps: AlertModalProps = {
-  alertType: "WARN",
-  title: "",
-  buttonTexts: {
-    primaryActionText: "Back",
-    secondaryActionText: "Confirm"
-  },
-  visible: false,
-  onOk: () => {},
-  onCancel: () => {},
-  onExit: () => {}
+const getTranslationKeyFromError = (error: Error): string => {
+  return (
+    errorNameToTranslationKeyMappings[error.name] ??
+    getTranslationKeyFromErrorMessage(error.message)
+  );
 };
 
-export const defaultConfirmationProps: AlertModalProps = {
-  alertType: "CONFIRM",
+const messageToTranslationKeyMappings: Record<string, string> = {
```

Comment:
Is it possible to not have this mapping, and instead store the keys in a map or enum called ERROR_MESSAGE instead? Right now, it seems like we are getting the enum which is the eng translation, getting the key, then getting the translation again. I think it would make more sense if we just use the key directly throughout our application and use that to get the translation!

Reply:
We had this discussion previously on Slack~ Will send you link on Slack PM